src/components/Payment.js

```diff
@@ -0,0 +1,69 @@
+import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+
+class Payment extends Component {
+  constructor(props) {
+    super(props);
+    this.calculateBalance = this.calculateBalance.bind(this);
+    this.renderAmountField = this.renderAmountField.bind(this);
+    this.state = {
+      balance: props.startingBalance,
+      amountInputs: []
+    };
+  }
+
+  calculateBalance() {
+    const paid = this.state.amountInputs.reduce((total, input) => {
+      return total + parseFloat(input.value || 0);
+    }, 0);
+
+    this.setState({ balance: this.props.startingBalance - paid });
+  }
+
+  componentDidMount() {
+    const amountInputs = Array.from(this.refs.form.querySelectorAll('input'));
```

Comment:
I don't know how I feel about this and `calculateBalance`.. not sure how else to keep a reference to all (both) of the input fields in the form. Would rather not have local state for the input values separately.. so I decided to calculate total on the fly on every change (can probably also throttle that..). Any thoughts?